pkg/rbac/middleware.go

```diff
@@ -0,0 +1,102 @@
+package rbac
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/stolostron/search-v2-api/pkg/config"
+	authv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+)
+
+var kClientset *kubernetes.Clientset
+
+func KubeClient() {
+	config, err := config.GetClientConfig()
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+	kClientset, err = kubernetes.NewForConfig(config)
+
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+
+}
+
+//verifies token (userid) with the TokenReview:
+func Middleware() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+			// Read the value of the client identifier from the request header
+			fmt.Fprintf(w, "%s %s %s \n", r.Method, r.URL, r.Proto)
+
+			//if there is cookie in header available use that else use the authorization:
+			var tokenId string
+			if len(r.Cookies()) != 0 {
+				for _, cookie := range r.Cookies() {
+					if cookie.Name == "acm-access-token-cookie" {
+						log.Println("Cookie: ", cookie.Value)
```

Comment:
- We can't log the token per secure engineer requirements, but we should log that we found the token in the cookie.
- Use klog for consistency.
- Use a higher verbosity level, to avoid logging too much.
```suggestion
						klog.V(5).Info("Got user token from cookie.")
```

Reply:
Okay. Do we want to log the API TokenReview result?
ex:
```
{
        "authenticated": true,
        "user": {
                "username": "kube:admin",
                "groups": [
                        "system:cluster-admins",
                        "system:authenticated"
                ],
                "extra": {
                        "scopes.authorization.openshift.io": [
                                "user:full"
                        ]
                }
        },
        "audiences": [
                "https://kubernetes.default.svc"
        ]
}
```

Reply:
We can log it if you find it helpful for debugging problems later, but certainly use a higher verbosity level. **Hint**, you could separate it in 2 different logs with different verbosity. Example:
```go
klog.V(5).Info("Got user token from cookie.")
klog.V(9).Info("TokenReview result: %s", resultData)
```